library/src/main/java/net/sparkminds/library/service/impl/HandleCSVImpl.java

```diff
@@ -0,0 +1,267 @@
+package net.sparkminds.library.service.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.opencsv.CSVReader;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import net.sparkminds.library.dto.bookmana.BookManaResponse;
+import net.sparkminds.library.entity.Author;
+import net.sparkminds.library.entity.Book;
+import net.sparkminds.library.entity.Category;
+import net.sparkminds.library.enumration.EnumLanguage;
+import net.sparkminds.library.enumration.EnumStatus;
+import net.sparkminds.library.exception.RequestException;
+import net.sparkminds.library.repository.AuthorRepository;
+import net.sparkminds.library.repository.BookRepository;
+import net.sparkminds.library.repository.CategoryRepository;
+import net.sparkminds.library.service.HandleCSVService;
+import net.sparkminds.library.service.criteria.BookCriteria;
+import net.sparkminds.library.service.query.BookQueryService;
+import net.sparkminds.library.util.FileUtil;
+import tech.jhipster.service.filter.StringFilter;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class HandleCSVImpl implements HandleCSVService{
```

Comment:
a thấy cái tên service với function trong nó ko liên quan lắm ta. Em sửa cách đặt tên lại nh người khác nhìn vào tưởng đâu đây là service handle common cho toàn bộ file CSV